compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -179,6 +200,19 @@ impl<'a> Parser<'a> {
         &self.sess.span_diagnostic
     }
 
+    pub(super) fn restore(&mut self, snapshot: SnapshotParser<'a>) {
```

Comment:
```suggestion
    pub(super) fn restore_snapshot(&mut self, snapshot: SnapshotParser<'a>) {
```

Similarly, a documentation comment would be helpful.